R/eval-tidy.R

```diff
@@ -251,16 +295,19 @@ delayedAssign(".data", as_data_pronoun(list()))
 #'
 #' # Now we can reference the values with the pronouns:
 #' eval_tidy(quote(.fns$c(.data$a, .data$b, .data$c)), data = mask)
-#'
-#'
-#' # Passing a data mask built with an explicit `top` and `bottom` is
-#' # not the same as passing a simple environment to `eval_tidy()`. In
-#' # the latter case, the ancestry of the environment is entirely
-#' # ignored. The following does not work because `+` is defined in an
-#' # upper level:
-#' try(eval_tidy(quote(a + b), data = bottom))
-as_data_mask <- function(data, parent = base_env()) {
-  .Call(rlang_as_data_mask, data, parent)
+as_data_mask <- function(data, parent = NULL) {
+  if (!is_null(parent)) {
+    id <- "rlang::as_data_mask() - parent argument"
+    msg <- paste(sep = "\n",
+      "The `parent` argument of `as_data_mask()` is deprecated.",
+      "The parent of the data mask is determined from either:\n",
```

Comment:
I think this would be clearer if you added an extra `""` and got rid of the `\n` (and maybe defined `paste_lines()` in utils